? .DS_Store
? dist
? aolrtf/bin
? app/bin
? common/bin
? oscar/bin
? oscar/src/net/kano/.DS_Store
? oscar/src/net/kano/joustsim/.DS_Store
? oscar/src/net/kano/joustsim/oscar/oscar/service/.DS_Store
Index: common/src/net/kano/joustsim/JavaTools.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/common/src/net/kano/joustsim/JavaTools.java,v
retrieving revision 1.2
diff -u -r1.2 JavaTools.java
--- common/src/net/kano/joustsim/JavaTools.java	5 Nov 2005 05:15:28 -0000	1.2
+++ common/src/net/kano/joustsim/JavaTools.java	24 Dec 2005 04:07:17 -0000
@@ -69,7 +69,7 @@
     public static <L> L getDelegatingProxy(
             final Collection<? extends L> listeners,
             Class<? extends L> cls) {
-        return cls.cast(Proxy.newProxyInstance(cls.getClassLoader(),
+        return  (L) (Proxy.newProxyInstance(cls.getClassLoader(),
                 new Class<?>[]{cls},
                 new InvocationHandler() {
                     public Object invoke(Object proxy, Method method,
Index: oscar/src/net/kano/joustsim/oscar/AimConnection.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/AimConnection.java,v
retrieving revision 1.14
diff -u -r1.14 AimConnection.java
--- oscar/src/net/kano/joustsim/oscar/AimConnection.java	5 Nov 2005 05:13:36 -0000	1.14
+++ oscar/src/net/kano/joustsim/oscar/AimConnection.java	24 Dec 2005 04:07:17 -0000
@@ -287,7 +287,7 @@
     private @Nullable <E extends Service> E getServiceOfType(int fam, Class<E> cls) {
         Service service = getService(fam);
         if (cls.isInstance(service)) {
-            return cls.cast(service);
+            return (E) service;
         } else {
             return null;
         }
Index: oscar/src/net/kano/joustsim/oscar/BuddyInfoManager.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/BuddyInfoManager.java,v
retrieving revision 1.8
diff -u -r1.8 BuddyInfoManager.java
--- oscar/src/net/kano/joustsim/oscar/BuddyInfoManager.java	16 Sep 2005 16:47:55 -0000	1.8
+++ oscar/src/net/kano/joustsim/oscar/BuddyInfoManager.java	24 Dec 2005 04:07:17 -0000
@@ -66,8 +66,10 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.logging.Logger;
 
 public class BuddyInfoManager {
+	private static final Logger LOGGER = Logger.getLogger(BuddyInfoManager.class.getName());
     private final AimConnection conn;
     private Map<Screenname,BuddyInfo> buddyInfos = new HashMap<Screenname, BuddyInfo>();
     private Map<BuddyHashHolder,BuddyCertificateInfo> cachedCertInfos
@@ -176,7 +178,7 @@
 
             public void handleCertificateInfo(InfoService service, Screenname buddy,
                     BuddyCertificateInfo certInfo) {
-                System.out.println("BuddyInfoManager got cert info for " + buddy);
+                LOGGER.fine("BuddyInfoManager got cert info for " + buddy);
                 BuddyInfo buddyInfo = getBuddyInfoInstance(buddy);
                 if (certInfo != null) cacheCertInfo(certInfo);
                 buddyInfo.setCertificateInfo(certInfo);
Index: oscar/src/net/kano/joustsim/oscar/ExternalServiceManager.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/ExternalServiceManager.java,v
retrieving revision 1.2
diff -u -r1.2 ExternalServiceManager.java
--- oscar/src/net/kano/joustsim/oscar/ExternalServiceManager.java	5 Nov 2005 05:14:22 -0000	1.2
+++ oscar/src/net/kano/joustsim/oscar/ExternalServiceManager.java	24 Dec 2005 04:07:17 -0000
@@ -107,7 +107,7 @@
     }
 
     private Timer initializeServiceTimer() {
-        Timer serviceTimer = new Timer("Service connection timeout watcher", true);
+        Timer serviceTimer = new Timer(true);
         serviceTimer.scheduleAtFixedRate(new TimerTask() {
             public void run() {
                 //TODO: use configurable connection timeout
@@ -182,7 +182,7 @@
         ServiceArbiter<?> arbiter = getServiceArbiter(family);
         if (arbiterClass.isInstance(arbiter)) {
             //noinspection unchecked
-            return arbiterClass.cast(arbiter);
+            return (A) (arbiter);
         } else {
             return null;
         }
Index: oscar/src/net/kano/joustsim/oscar/oscar/service/AbstractServiceArbiter.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/oscar/service/AbstractServiceArbiter.java,v
retrieving revision 1.1
diff -u -r1.1 AbstractServiceArbiter.java
--- oscar/src/net/kano/joustsim/oscar/oscar/service/AbstractServiceArbiter.java	5 Nov 2005 05:12:40 -0000	1.1
+++ oscar/src/net/kano/joustsim/oscar/oscar/service/AbstractServiceArbiter.java	24 Dec 2005 04:07:18 -0000
@@ -138,7 +138,7 @@
         for (ServiceArbiterRequest request : requests) {
             if (cls.isInstance(request)) {
                 if (found == null) {
-                    found = cls.cast(request);
+                    found = (E) (request);
                 } else {
                     throw new IllegalArgumentException("Multiple instances of "
                             + cls + " in requests: " + requests);
Index: oscar/src/net/kano/joustsim/oscar/oscar/service/icbm/Conversation.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/oscar/service/icbm/Conversation.java,v
retrieving revision 1.2
diff -u -r1.2 Conversation.java
--- oscar/src/net/kano/joustsim/oscar/oscar/service/icbm/Conversation.java	26 Jun 2005 16:46:51 -0000	1.2
+++ oscar/src/net/kano/joustsim/oscar/oscar/service/icbm/Conversation.java	24 Dec 2005 04:07:18 -0000
@@ -40,7 +40,10 @@
 import net.kano.joscar.MiscTools;
 import net.kano.joustsim.Screenname;
 
+import java.util.logging.Logger;
+
 public abstract class Conversation {
+	private static final Logger LOGGER = Logger.getLogger(Conversation.class.getName());
     private final Screenname buddy;
 
     private CopyOnWriteArrayList<ConversationListener> listeners
@@ -126,7 +129,7 @@
 
         DefensiveTools.checkNull(event, "event");
 
-        System.out.println(MiscTools.getClassName(this) + ": firing incoming "
+        LOGGER.fine(MiscTools.getClassName(this) + ": firing incoming "
                 + "event: " + event);
         if (event instanceof MessageInfo) {
             MessageInfo messageInfo = (MessageInfo) event;
@@ -145,7 +148,7 @@
 
         DefensiveTools.checkNull(event, "event");
 
-        System.out.println(MiscTools.getClassName(this) + ": firing outgoing "
+        LOGGER.fine(MiscTools.getClassName(this) + ": firing outgoing "
                 + "event: " + event);
         if (event instanceof MessageInfo) {
             MessageInfo messageInfo = (MessageInfo) event;
Index: oscar/src/net/kano/joustsim/oscar/oscar/service/info/BuddyTrustManager.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/oscar/service/info/BuddyTrustManager.java,v
retrieving revision 1.3
diff -u -r1.3 BuddyTrustManager.java
--- oscar/src/net/kano/joustsim/oscar/oscar/service/info/BuddyTrustManager.java	5 Nov 2005 05:13:01 -0000	1.3
+++ oscar/src/net/kano/joustsim/oscar/oscar/service/info/BuddyTrustManager.java	24 Dec 2005 04:07:18 -0000
@@ -48,8 +48,10 @@
 import java.beans.PropertyChangeEvent;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Logger;
 
 public class BuddyTrustManager {
+	private static final Logger LOGGER = Logger.getLogger(BuddyTrustManager.class.getName());
     private final BuddyInfoManager buddyInfoManager;
 
     private final AimConnection conn;
@@ -86,7 +88,7 @@
                 if (prop.equals(BuddyInfo.PROP_CERTIFICATE_INFO)) {
                     BuddyCertificateInfo certInfo
                             = (BuddyCertificateInfo) event.getNewValue();
-                    System.out.println("cert info for " + buddy + " changed: " + certInfo);
+                    LOGGER.fine("cert info for " + buddy + " changed: " + certInfo);
                     if (certInfo != null) {
                         // we want to cache this certificate before we handle
                         // the buddy's new hash
@@ -191,9 +193,9 @@
             // trust status changed
             oldStatus = buddyTrustInfo.getTrustedStatus();
 
-            System.out.println("updating buddy hash for " + sn);
+            LOGGER.fine("updating buddy hash for " + sn);
             newStatus = updateBuddyHash(sn, certInfo);
-            System.out.println("new status for " + sn + " is " + newStatus);
+            LOGGER.fine("new status for " + sn + " is " + newStatus);
             if (newStatus == null) {
                 // the status did not change, so there's nothing left for us to
                 // do here
Index: oscar/src/net/kano/joustsim/oscar/oscar/service/info/CertificateInfoRequestManager.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/oscar/service/info/CertificateInfoRequestManager.java,v
retrieving revision 1.2
diff -u -r1.2 CertificateInfoRequestManager.java
--- oscar/src/net/kano/joustsim/oscar/oscar/service/info/CertificateInfoRequestManager.java	26 Jun 2005 16:46:51 -0000	1.2
+++ oscar/src/net/kano/joustsim/oscar/oscar/service/info/CertificateInfoRequestManager.java	24 Dec 2005 04:07:18 -0000
@@ -46,14 +46,17 @@
 
 import java.security.cert.X509Certificate;
 import java.util.Iterator;
+import java.util.logging.Logger;
 
 public class CertificateInfoRequestManager extends UserInfoRequestManager {
+	private static final Logger LOGGER = Logger.getLogger(CertificateInfoRequestManager.class.getName());
+
     public CertificateInfoRequestManager(InfoService service) {
         super(service);
     }
 
     protected SnacCommand generateSnacCommand(final Screenname sn) {
-        System.out.println("generating cert request for " + sn);
+        LOGGER.fine("generating cert request for " + sn);
         return new GetInfoCmd(GetInfoCmd.FLAG_CERT, sn.getFormatted());
     }
 
Index: oscar/src/net/kano/joustsim/oscar/oscar/service/ssi/AbstractGroup.java
===================================================================
RCS file: /cvsroot/joustsim/joustsim/oscar/src/net/kano/joustsim/oscar/oscar/service/ssi/AbstractGroup.java,v
retrieving revision 1.2
diff -u -r1.2 AbstractGroup.java
--- oscar/src/net/kano/joustsim/oscar/oscar/service/ssi/AbstractGroup.java	14 Jul 2005 04:37:02 -0000	1.2
+++ oscar/src/net/kano/joustsim/oscar/oscar/service/ssi/AbstractGroup.java	24 Dec 2005 04:07:19 -0000
@@ -40,12 +40,12 @@
 import java.util.Collection;
 import java.util.Set;
 import java.util.HashSet;
-import java.util.concurrent.CopyOnWriteArrayList;
+//import java.util.concurrent.CopyOnWriteArrayList;
 
 abstract class AbstractGroup implements Group {
     private final SimpleBuddyList buddyList;
-    private CopyOnWriteArrayList<GroupListener> listeners
-            = new CopyOnWriteArrayList<GroupListener>();
+    private ArrayList<GroupListener> listeners
+            = new ArrayList<GroupListener>();
 
     private List<SimpleBuddy> buddies = new ArrayList<SimpleBuddy>();
 
@@ -116,12 +116,13 @@
         }
     }
 
-    protected CopyOnWriteArrayList<GroupListener> getListeners() {
+    protected ArrayList<GroupListener> getListeners() {
         return listeners;
     }
 
     public void addGroupListener(GroupListener listener) {
-        listeners.addIfAbsent(listener);
+    	if (!listeners.contains(listener))
+	        listeners.add(listener);
     }
 
     public void removeGroupListener(GroupListener listener) {
